File: src/taxes/quebec-parental-insurance-plan.ts
/*
Sources
    https://www.rqap.gouv.qc.ca/fr/a-propos-du-regime/information-generale/cotisations-et-revenu-maximal-assurable

Revised
    2025-12-22
*/

export interface PremiumRates {
    SELF_EMPLOYED: number;
    SALARIED: number;
}

export interface QuebecParentalInsurancePlan {
    MAX_INSURABLE_EARNINGS: number;
    PREMIUM_RATES: PremiumRates;
}

export const QPIP: QuebecParentalInsurancePlan = {
    MAX_INSURABLE_EARNINGS: 103000,
    PREMIUM_RATES: {
        SELF_EMPLOYED: 0.00764,
        SALARIED: 0.0043,
    },
};

File: src/commonMain/kotlin/com/equisoft/taxca/taxes/QuebecParentalInsurancePlan.kt
/*
Sources
    https://www.rqap.gouv.qc.ca/fr/a-propos-du-regime/information-generale/cotisations-et-revenu-maximal-assurable

Revised
    2025-12-22
*/
package com.equisoft.taxca.taxes

data class PremiumRates(
    val selfEmployed: Double,
    val salaried: Double,
)

data class QuebecParentalInsurancePlan(
    val maxInsurableEarnings: Double,
    val premiumRates: PremiumRates,
)

val Qpip: QuebecParentalInsurancePlan = QuebecParentalInsurancePlan(
    maxInsurableEarnings = 103000.0,
    premiumRates = PremiumRates(
        selfEmployed = 0.00764,
        salaried = 0.0043,
    ),
)
